### PR TITLE
[tree] doxygen warnings

### PR DESCRIPTION
**Pull request by ferdymercury**
# This Pull request:

## Changes or fixes:
Several documentation errors found by doxygen

## Checklist:

- [x] tested changes locally
- [x] updated the docs (if necessary)

### CODE REVIEW

File: tree/treeplayer/inc/TTreeIndex.h
Comment (pcanal) [inline tree/treeplayer/inc/TTreeIndex.h:74]: @couet @ferdymercury Can you verify that this type of change preserve the value of TClass::GetTitle:
```
root [1] TClass::GetClass("TTreeIndex")->GetTitle()
(const char *) "A Tree Index with majorname and minorname."
```

**Review by couet — APPROVED** [review]
The doxygen fixes look good to me. I let @eguiraud verify the code changes.

File: tree/treeplayer/inc/TTreeIndex.h
Comment (couet) [reply tree/treeplayer/inc/TTreeIndex.h:74]: @ferdymercury can you verify that as your root version has this change ?

Comment (ferdymercury) [reply tree/treeplayer/inc/TTreeIndex.h:74]: Hmm I suspect that it won't preserve it, but I will check it. Doxygen doesn't seem to like however the original formulation.

If it does not preserve it: would it be ok to rename those macros to:
`ClassDefTitle(TTreeIndex,2,"A Tree Index with majorname and minorname")`

and to store the title directly via this new macro, rather than via a comment ? (It would be more type-safe I guess).

Comment (ferdymercury) [reply tree/treeplayer/inc/TTreeIndex.h:74]: Never mind, it does not preserve it, so will revert those changes.

**Review by eguiraud — APPROVED** [review]
Thank you very much @ferdymercury , this is great. Just one thing I don't understand below.

File: tree/tree/src/TChain.cxx
Comment (ferdymercury) [reply tree/tree/src/TChain.cxx:301]: yes, an # means a hyperlink in doxygen/markdown documentation. To actually 'show' an #, one can escape it using \# or #%

Comment (eguiraud) [reply tree/tree/src/TChain.cxx:301]: got it, thanks!

**Comment by eguiraud** [comment]
@phsft-bot build

**Comment by eguiraud** [comment]
Before merging, @pcanal are the semicolon removals after `ClassDef` desirable or not here? (asking because of https://github.com/root-project/root/pull/9655#discussion_r789840944 )

**Comment by ferdymercury** [comment]
> Before merging, @pcanal are the semicolon removals after `ClassDef` desirable or not here? (asking because of [#9655 (comment)](https://github.com/root-project/root/pull/9655#discussion_r789840944) )

I'll revert them, as we concluded that they were rather a 'bug' from doxygen.

**Comment by ferdymercury** [comment]
Fixed

**Comment by eguiraud** [comment]
Alright, thanks @ferdymercury ! Please rebase on the latest master to fix conflicts (you can also take this change to squash all commits into one). @pcanal I'll merge when this is done unless you have something against it.

**Comment by ferdymercury** [comment]
Thanks egiraud, I fixed the conflict. However I did not find the option to squash the commits when resolving via the GUI. I believe this can be done once you merge, right?

**Comment by eguiraud** [comment]
Something went wrong, it looks like you merged the master branch into your feature branch. We need to have one (or more) commit of yours on top of the master branch, like this:

image

Instead the current situation is:

image

Usually I use `git rebase --interactive` from the command line to fix these kind of issues, see e.g. https://www.sitepoint.com/git-interactive-rebase-guide/ .

To display the state of the repository I use a custom git log command that basically runs `git log --graph --decorate --oneline --abbrev-commit`.

**Comment by ferdymercury** [comment]
ohh I see. I have to teach a class now, will fix it later this evening.

**Comment by ferdymercury** [comment]
I tried rebase but it didn't look good either, so I just made a new PR here https://github.com/root-project/root/pull/9678

**Pull request closed without merging**
